dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/DatabaseIdentityStore.java

```diff
@@ -166,9 +164,15 @@ public Set<String> getCallerGroups(CredentialValidationResult validationResult)
         if (!validationTypes().contains(ValidationType.PROVIDE_GROUPS)) {
             return groups;
         }
-        SecurityManager securityManager = System.getSecurityManager();
-        if (securityManager != null) {
-            securityManager.checkPermission(new IdentityStorePermission(JavaEESecConstants.GET_GROUPS_PERMISSION));
+
+        // Get the service through OSGi
+        IdentityStorePermissionService permissionService = CDIHelper.getService(IdentityStorePermissionService.class);
+        if (permissionService == null) {
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, "IdentityStorePermissionService not found via OSGi, skipping check.");
+            }
+        } else {
+            permissionService.checkGetGroupsPermission();
```

Comment:
```suggestion

        IdentityStorePermissionService.checkPermission(JavaEESecConstants.GET_GROUPS_PERMISSION);
```
Will need to remove the extra `}` not in this suggestion because it is not part of the change.

Reply:
Yes, have done as you've suggested.